src/NHibernate/IFutureValueAsync.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	public interface IFutureValueAsync<T>
+	{
+		Task<T> GetValue();
```

Comment:
I wonder if we could avoid those additional future methods when not supplying a cancellation token to them: would it be possible to simply add the async `GetValue` on the currently synchronous `IFutureValue`? For `IEnumerable` yielded by `ToFuture`, we will still need an `Async` variant, but the underlying implementation would probably be lighter if we could here too just implement th `IAsyncEnumerable` interface on top of the current synchronous enumerable internal class.